app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,44 @@
+class CartsController < ApplicationController
+  before_action :check_product, only: :add_to_cart
+
+  def add_to_cart
+    if logged_in?
+      flash[:success] = t("cart.add_item", name: @product.name)
+      handle_quantity
+      session[:cart] << @product.id
+      redirect_to root_path
+    else
+      flash[:danger] = t "cart.must_login"
+      redirect_to login_path
+    end
+  end
+
+  def show
+    @products = Product.find session[:cart]
```

Comment:
ko dùng find em nhé, dùng where in đi em
vì find sẽ gây lỗi exception

Reply:
Chị ơi, em tìm không ra cú pháp where in trong rails
cơ mà where trong rails đi với array là sẽ sinh sql where in ngoài ra còn kiểm tra column có null hay không